package.json

```diff
@@ -17,55 +17,96 @@
     "test:hardhat": "hardhat test"
   },
   "resolutions": {
-    "@nomicfoundation/edr": "0.3.5",
     "ethers": "^5.7.2",
     "hardhat-deploy": "^0.12.1"
   },
   "devDependencies": {
-    "@babel/core": "^7.23.9",
-    "@fraxfinance/layerzero-v2-upgradeable": "git://github.com/fraxfinance/LayerZero-v2-upgradeable.git#v1.0.0",
-    "@layerzerolabs/lz-evm-messagelib-v2": "~2.3.21",
-    "@layerzerolabs/lz-evm-oapp-v2": "~2.3.19",
-    "@layerzerolabs/lz-evm-protocol-v2": "~2.3.3",
-    "@layerzerolabs/toolbox-foundry": "~0.1.5",
-    "@openzeppelin/contracts": "^4.8.1",
-    "@openzeppelin/contracts-upgradeable": "^4.8.1",
+    "@coral-xyz/anchor": "^0.29.0",
+    "@ethersproject/bytes": "^5.7.0",
+    "@layerzerolabs/devtools": "~0.4.6",
+    "@layerzerolabs/devtools-evm": "~1.0.5",
+    "@layerzerolabs/devtools-evm-hardhat": "^2.0.7",
+    "@layerzerolabs/devtools-solana": "~1.0.7",
+    "@layerzerolabs/eslint-config-next": "~2.3.39",
+    "@layerzerolabs/io-devtools": "~0.1.16",
+    "@layerzerolabs/lz-definitions": "^3.0.59",
+    "@layerzerolabs/lz-evm-messagelib-v2": "^3.0.12",
+    "@layerzerolabs/lz-evm-protocol-v2": "^3.0.12",
+    "@layerzerolabs/lz-evm-v1-0.7": "^3.0.12",
+    "@layerzerolabs/lz-solana-sdk-v2": "3.0.0",
+    "@layerzerolabs/lz-v2-utilities": "^3.0.12",
+    "@layerzerolabs/oapp-evm": "^0.3.1",
+    "@layerzerolabs/oft-evm": "^3.1.2",
+    "@layerzerolabs/oft-v2-solana-sdk": "^3.0.59",
+    "@layerzerolabs/prettier-config-next": "^2.3.39",
+    "@layerzerolabs/protocol-devtools": "^1.1.5",
+    "@layerzerolabs/protocol-devtools-evm": "~3.0.6",
+    "@layerzerolabs/protocol-devtools-solana": "^4.0.8",
+    "@layerzerolabs/solhint-config": "^3.0.12",
+    "@layerzerolabs/test-devtools-evm-foundry": "~6.0.2",
+    "@layerzerolabs/test-devtools-evm-hardhat": "~0.5.1",
+    "@layerzerolabs/toolbox-foundry": "~0.1.12",
+    "@layerzerolabs/toolbox-hardhat": "~0.6.7",
+    "@layerzerolabs/ua-devtools": "~3.0.5",
+    "@layerzerolabs/ua-devtools-evm": "~5.0.6",
+    "@layerzerolabs/ua-devtools-evm-hardhat": "~6.0.9",
+    "@layerzerolabs/ua-devtools-solana": "~4.1.1",
+    "@metaplex-foundation/mpl-token-metadata": "^3.2.1",
+    "@metaplex-foundation/mpl-toolbox": "^0.9.4",
+    "@metaplex-foundation/umi": "^0.9.2",
+    "@metaplex-foundation/umi-bundle-defaults": "^0.9.2",
+    "@metaplex-foundation/umi-eddsa-web3js": "^0.9.2",
+    "@metaplex-foundation/umi-public-keys": "^0.8.9",
+    "@metaplex-foundation/umi-web3js-adapters": "^0.9.2",
+    "@nomicfoundation/hardhat-ethers": "^3.0.5",
+    "@nomiclabs/hardhat-ethers": "^2.2.3",
+    "@nomiclabs/hardhat-waffle": "^2.0.6",
+    "@openzeppelin/contracts": "^5.0.2",
```

Comment:
This upgraded dependency will cause breaking changes in the OFT imports as the OFT uses 4.x.

Reply:
noted